### PR TITLE
BUILD_INFO: document the release process (Maven release plugin)

### PR DESCRIPTION
**Pull request by jouvin**

### CODE REVIEW

**Comment by stdweird** [comment]
test this please

File: BUILD_INFO.md [hunk 1]
```diff
@@ -97,3 +97,41 @@ files, particularly in this repository, be sure to understand Maven basics.
...
+[release](http://maven.apache.org/maven-release/maven-release-plugin/) plugin. The main goals are:
+
+* `release:prepare`: initialization of the release process. This produces a few files that allow to revert the release process.
+* `release:perform`: build and publish the new release
+
+The `release` plugin is taking care of updating the pom file, before and after deploying the release and doing a few other things (like
+tagging the repo). It can be viewed as sort of a wrapper over the `deploy` phase, which does the real deployment. It should be
+considered mandatory to use the `release` plugin to deploy releases.
+
+*Note: deploying a release requires to have GPG keys: create them if necessary before producing a release.*
```
Comment (jrha) [inline BUILD_INFO.md]: Does it also require a running `gpg-agent` like the main release process does?

File: BUILD_INFO.md [hunk 2]
```diff
@@ -97,3 +97,41 @@ files, particularly in this repository, be sure to understand Maven basics.
...
+We have already seen above how to update the *parent* information. Another useful plugin allows to updates
+the plugins used by the pom file: [versions](http://mojo.codehaus.org/versions-maven-plugin/). Particularly
+useful goals of this plugin are:
+
+* `versions:display-dependency-updates`: display dependencies for which a newer version exists
+* `versions:use-latest-versions`: update version of dependencies used to last version available
+* `versions:display-plugin-updates`: display used plugins for which a newer version exists. The version
+must be updated manually in the pom file if a new version exists and you want to use it.
+
+*Note: this is important to define the version to use for dependencies and plugins present in the configuration. Maven raises a
```
Comment (jrha) [inline BUILD_INFO.md]: `s/this is important/It is important/`

File: BUILD_INFO.md (hunk 1 above)
Comment (jouvin) [inline BUILD_INFO.md]: I guess so, like any `gpg` command I guess (I just discovered this requirement, I'll add it).

**Comment by jouvin** [comment]
Can we merge this PR, I'd like to see this improved doc part of the upcoming release (we can continue to improve it later).

**Comment by jouvin** [comment]
Wait one clarification on how to cancel a `release:prepare`. The current version is incorrect... I'll update it soon...

**Comment by jouvin** [comment]
Should be ok now!

**Pull request merged**